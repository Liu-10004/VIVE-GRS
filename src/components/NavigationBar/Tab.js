/**
 * @see https://github.com/vivedu/VIVEDU-Homepage/issues/60
 */

import React from 'react'
import { withRouter, Link } from 'react-router-dom'
import cx from 'classnames'
import styles from './Tab.scss'

export const RawTab = ({ children, active, link, ...rest }) => (
  <Link
    className={cx(styles.root, {
      [styles.active]: active,
    })}
    to={link}
    {...rest}
  >
    {children}
  </Link>
)

export default withRouter(RawTab)
