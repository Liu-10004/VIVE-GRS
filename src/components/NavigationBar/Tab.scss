@import '../../styles/colors.scss';

.root {
  display: flex;
  box-sizing: border-box;
  padding: 3px 6px 0;
  line-height: 1.2em;
  cursor: pointer;
  color: currentColor;

  & + & {
    margin-left: 34px;
  }

  &:hover {
    color: $themeColor;
  }
}

.active {
  // NOTE: 撑起下划线
  padding-bottom: 5px;
  color: $themeColor;
  border-bottom: 4px solid $themeColor;
}
