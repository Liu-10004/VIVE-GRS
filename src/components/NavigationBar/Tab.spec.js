import React from 'react'
import { StaticRouter } from 'react-router'
import { shallow } from 'enzyme'
import Tab from './Tab'

describe('<Tab /> with <NavigationBar />', () => {
  const context = {}

  it('should match old snapshot', () => {
    const wrapper = shallow(
      <StaticRouter context={context}>
        <Tab location={{ pathname: '' }} link="/about">
          About us
        </Tab>
      </StaticRouter>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
