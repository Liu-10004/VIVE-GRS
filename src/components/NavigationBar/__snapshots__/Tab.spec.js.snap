// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> with <NavigationBar /> should match old snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StaticRouter
    basename=""
    context={Object {}}
    location="/"
>
    <withRouter(RawTab)
        link="/about"
        location={
            Object {
                "pathname": "",
              }
        }
    >
        About us
    </withRouter(RawTab)>
</StaticRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withRouter(RawTab)
        link="/about"
        location={
                Object {
                        "pathname": "",
                      }
        }
>
        About us
</withRouter(RawTab)>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "About us",
        "link": "/about",
        "location": Object {
          "pathname": "",
        },
      },
      "ref": null,
      "rendered": "About us",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withRouter(RawTab)
          link="/about"
          location={
                    Object {
                              "pathname": "",
                            }
          }
>
          About us
</withRouter(RawTab)>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "About us",
          "link": "/about",
          "location": Object {
            "pathname": "",
          },
        },
        "ref": null,
        "rendered": "About us",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
